cEP-0010.md

```diff
@@ -0,0 +1,290 @@
+* `Root.Metadata.CommitMessage.Shortlog.FirstCharacter`: This aspect detects
+inconsistent casing for the first character of your commit message shortlog(
+and the the first character following the colon if there is one) given the
+value of
+`Root.Metadata.CommitMessage.Shortlog.FirstCharacter.shortlog_starts_upper_case`
+[(Taste obj) representing the case to use for the first character of your
+commit message shortlog and the first character following the colon].
+* `Root.Metadata.CommitMessage.Body`: Parent aspect for all of the aspects
+related to your commit message body.
+* `Root.Metadata.CommitMessage.Body.Existence`: Checks on the existence of
+your commit message body (it enforces its usage).
+* `Root.Metadata.CommitMessage.Body.Length`: Checks on the length of your
+commit message body given the value of
+`Root.Metadata.CommitMessage.Body.Length.max_body_length`(Taste obj).
+
+## Implementation
+
+As stated previously the aspects will be implemented in a tree like form,
+with the Root aspect at the root of the tree. Some classes, metaclasses
+and some few methods were define so far to make that possible including:
+
+* [the aspectbase class](https://github.com/coala/coala/blob/master/coalib/bearlib/aspects/base.py)
+* [the Documentation class for documenting aspects](https://github.com/coala/coala/blob/master/coalib/bearlib/aspects/docs.py)
+* [the aspectclass metaclass](https://github.com/coala/coala/blob/master/coalib/bearlib/aspects/meta.py)
+* [the TasteMeta class](https://github.com/coala/coala/blob/master/coalib/bearlib/aspects/taste.py)
+* [the Taste class](https://github.com/coala/coala/blob/master/coalib/bearlib/aspects/taste.py)
+
+`aspects` will simply be defined this way:
+```python
+from coalib.bearlib.aspects import Root, Taste
+
+
+@Root.subaspect
+class SomeAspect:
+#This will serve as documentation for the aspect
+    class docs:
+        example = 'an example of this aspect of code'
+        example_language = 'the programming language of the example'
+        importance_reason = 'the reason why this is important'
+        fix_suggestions = ''
+#Here we define the different tastes for this aspect
+#aspects are not compelled to have tastes
+    some_taste = Taste[Taste_type](
+        'Documentation about this taste',
+        list('Possible value for this taste',),
+        default='Default value for this taste')
+```
+The documentation of each of the implemented aspects will be added to
+the [coala/aspect-docs](https://github.com/coala/aspect-docs) repository.
+
+Concerning bears' results annotation, so far, the `aspect` attr was added to
+the [Result class](https://github.com/coala/coala/blob/master/coalib/results/Result.py)
+(where a Result object is what is yielded by a bear when through executing).
+So annotating a bear's result with aspects simply consists of setting this
+variable to the appropriate aspect of code. The `CAN_DETECT` and `CAN_FIX` bears' attr
+will be changed to `DETECT_ASPECTS` and `FIX_ASPECTS`.
```

Comment:
Forgot that we threw the `DETECT_ASPECTS` and `FIX_ASPECTS` attributes away in favor of the metaclass based aspects definition? ;)

https://github.com/coala/cEPs/pull/72/commits/9aa3b43fa882e247a9fd15cea02c10614ca986cf#diff-7c040d00c1a93d80315fca153d104cdaR100